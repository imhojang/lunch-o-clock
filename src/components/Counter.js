import React from 'react';

const Counter = ({ count, decrement, increment }) => {
  return (
    <div>
      <button onClick={decrement}>-</button>
      {count}
      <button onClick={increment}>+</button>
    </div>
  );
};

export default Counter;
