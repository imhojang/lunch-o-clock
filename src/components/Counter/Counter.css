.counter-container > input {
  width: 100px;
  display: inline-block;
  font-size: 20px;
  text-align:center;
  background: none;
  border: none;
}

.counter-container > input::-webkit-outer-spin-button,
.counter-container > input::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.counter-container > input[type=number] {
  -moz-appearance:textfield;
}

.counter-container > svg {
  cursor: pointer;
  width: 15px;
  vertical-align: sub;
}

.counter-container > svg:active {
  fill: #fce285;
}
