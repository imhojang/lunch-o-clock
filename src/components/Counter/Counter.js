import React from 'react'
import PropTypes from 'prop-types'
import './Counter.css'
import { IncrementIcon, DecrementIcon } from '../../utils/Icons'

const Counter = ({ count, decrement, increment, updateCount }) => {
  return (
    <div className='counter-container'>
      <DecrementIcon className='decrement-button' handleClick={decrement} />
      <input
        className='count'
        name='count-input'
        type='number'
        value={count}
        onChange={e => updateCount(e.target.value)}
      />
      <IncrementIcon className='increment-button' handleClick={increment} />
    </div>
  )
}

Counter.propTypes = {
  count: PropTypes.number.isRequired,
  increment: PropTypes.func.isRequired,
  decrement: PropTypes.func.isRequired,
  updateCount: PropTypes.func.isRequired
}

export default Counter
