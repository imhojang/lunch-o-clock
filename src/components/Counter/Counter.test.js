import React from 'react'
import { mount, shallow } from 'enzyme'
import Counter from './Counter'

describe('Counter structure', () => {
  let counter
  let count = 2
  let decrement = jest.fn()
  let increment = jest.fn()
  let props = { count, decrement, increment }

  it('renders correctly', () => {
    counter = shallow(<Counter {...props} />)
  })

  it('matches snapshot', () => {
    expect(counter).toMatchSnapshot()
  })

  it('renders increment button correctly', () => {
    const incrementButton = counter.find('.increment-button')
    expect(incrementButton).toBeDefined()
  })

  it('renders decrement button correctly', () => {
    const decrementButton = counter.find('.decrement-button')
    expect(decrementButton).toBeDefined()
  })

  it('renders count display correctly', () => {
    const countDisplay = counter.find('.count')
    expect(countDisplay).toBeDefined()
  })
})

describe('Counter behavior', () => {
  let counter
  let count
  let decrement
  let increment

  beforeEach(() => {
    count = 3
    decrement = jest.fn()
    increment = jest.fn()
    let props = { count, decrement, increment }
    counter = mount(<Counter {...props} />)
  })

  it('props are passed down correctly', () => {
    expect(counter.props().count).toEqual(3)
    expect(counter.props().decrement).toBe(decrement)
    expect(counter.props().increment).toBe(increment)
  })

  it('check if decrement button calls decrement correctly', () => {
    const decrementButton = counter.find('.decrement-button')
    decrementButton.simulate('click')
    expect(decrement).toHaveBeenCalled()
  })

  it('check if increment button calls increment correctly', () => {
    const incrementButton = counter.find('.increment-button')
    incrementButton.simulate('click')
    expect(increment).toHaveBeenCalled()
  })
})
