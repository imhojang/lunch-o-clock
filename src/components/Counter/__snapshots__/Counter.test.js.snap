// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter structure matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Counter
    count={2}
    decrement={[Function]}
    increment={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown
          className="decrement-button"
          handleClick={[Function]}
/>,
        <input
          className="count"
          name="count-input"
          onChange={[Function]}
          value={2}
/>,
        <Unknown
          className="increment-button"
          handleClick={[Function]}
/>,
      ],
      "className": "counter-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "decrement-button",
          "handleClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "count",
          "name": "count-input",
          "onChange": [Function],
          "value": 2,
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "increment-button",
          "handleClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            className="decrement-button"
            handleClick={[Function]}
/>,
          <input
            className="count"
            name="count-input"
            onChange={[Function]}
            value={2}
/>,
          <Unknown
            className="increment-button"
            handleClick={[Function]}
/>,
        ],
        "className": "counter-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "decrement-button",
            "handleClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "count",
            "name": "count-input",
            "onChange": [Function],
            "value": 2,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "increment-button",
            "handleClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
